src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml.cs

```diff
@@ -741,7 +741,15 @@ private void ItemSelected_Checked(object sender, RoutedEventArgs e)
 		private void ItemSelected_Unchecked(object sender, RoutedEventArgs e)
 		{
 			if (sender is CheckBox checkBox && checkBox.DataContext is ListedItem item && FileList.SelectedItems.Contains(item))
+			{
+				FileList.SelectionChanged -= FileList_SelectionChanged;
+
 				FileList.SelectedItems.Remove(item);
+				if (SelectedItems is not null && SelectedItems.Contains(item))
+					SelectedItems.Remove(item);
+
+				FileList.SelectionChanged += FileList_SelectionChanged;
+			}
```

Comment:
Instead of trying to try ignore the selection event when unchecking a checkbox, I think the proper solution would be to introduce a new VSM VisualState Group for the selection states and switch between the "Selected" and "Unselected" while also maintaining a "Hover" and "Normal" state.

Reply:
Good idea.

Reply:
@chingucoding I don't know much about VSMs. Can you create a new pr to fix this bug?

Reply:
Sure if you prefer that I can create a new PR to fix that.